src/analyses/mutexEventsAnalysis.ml

```diff
@@ -24,15 +24,15 @@ struct
     match lv_opt with
     | None ->
       Queries.AD.iter (fun e ->
-          ctx.split () [Events.Lock (e, rw)]
+          ctx.split () [Events.Lock (e, rw); Events.RefinePointerExp {exp = arg; ad = Queries.AD.singleton e}]
```

Comment:
Why is a new special-purpose refinement event needed? Doesn't the general `SplitBranch` one work? Basically to `branch` on `arg == e` (or `arg == &e` maybe?). It's a singleton set, so it should work.